src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -148,9 +148,466 @@
+        {
+            NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];            
+            numberFormatter.locale = currentLocale;
+            numberFormatter.numberStyle = NSNumberFormatterPercentStyle;
+            static const char* Patterns[] = {"-n %", "-n%", "-%n", "%-n", "%n-", "n-%", "n%-", "-% n", "n %-", "% n-", "% -n", "n- %"};
+            const char *pFormat = [[numberFormatter negativeFormat] UTF8String];
+            value = GetNumericPatternNative(pFormat, Patterns, sizeof(Patterns)/sizeof(Patterns[0]), true);
+            if (value >= 0)
+            {
+                return value;
+            }
+            else
+            {
+                isSuccess = false;
+            }
+            break;
+        }
+        case LocaleNumber_PositivePercentFormat:
+        {
+            NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];            
+            numberFormatter.locale = currentLocale;
+            numberFormatter.numberStyle = NSNumberFormatterPercentStyle;
+            static const char* Patterns[] = {"n %", "n%", "%n", "% n"};
+            const char *pFormat = [[numberFormatter positiveFormat] UTF8String];
+            value = GetNumericPatternNative(pFormat, Patterns, sizeof(Patterns)/sizeof(Patterns[0]), false);
+            if (value >= 0)
+            {
+                return value;
+            }
+            else
+            {
+                isSuccess = false;
+            }
+            break;
+        }
+        default:
+            value = -1;
+            assert(isSuccess);
+            break;
+    }
+
+    return value;
+}
+
+/*
+PAL Function:
+GlobalizationNative_GetLocaleInfoPrimaryGroupingSizeNative
+
+Returns primary grouping size for decimal and currency
+*/
+int32_t GlobalizationNative_GetLocaleInfoPrimaryGroupingSizeNative(const char* localeName, LocaleNumberData localeGroupingData)
+{
+    bool isSuccess = true;
+    NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+    NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+    NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];            
+    numberFormatter.locale = currentLocale;    
+
+    switch (localeGroupingData)
+    {
+        case LocaleNumber_Digit:
+            numberFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+            break;
+        case LocaleNumber_Monetary:
+            numberFormatter.numberStyle = NSNumberFormatterCurrencyStyle;
+            break;
+        default:
+            isSuccess = false;
+            assert(isSuccess);
+            break;
+    }
+    return [numberFormatter groupingSize];    
+}
+
+/*
+PAL Function:
+GlobalizationNative_GetLocaleInfoSecondaryGroupingSizeNative
+
+Returns secondary grouping size for decimal and currency
+*/
+int32_t GlobalizationNative_GetLocaleInfoSecondaryGroupingSizeNative(const char* localeName, LocaleNumberData localeGroupingData)
```

Comment:
What are the expected return values for this? Is 0 ever returned? If its not different than the primary group sizing, will it just return the same value as the primary group sizing.

If 0 isn't returned then this wouldn't be hit
```
            if (secondaryGroupingSize == 0)
            {
                return new int[] { primaryGroupingSize };
            }
```

Reply:
It is different from primary group sizing as this returns [numberFormatter secondaryGroupingSize] and for primary it will return [numberFormatter groupingSize]. It can be 0 depending on the culture if [numberFormatter secondaryGroupingSize] is 0;

Reply:
Are there cultures that have a 0 secondary grouping size?
That would mean there are numbers like 123...123456,789.00

I saw from some docs that if the property is left blank (I guess unspecified?) it would have a value that is the same as the primary grouping size. This leads me to think that `secondaryGroupingSize` won't be zero.

Reply:
My bad I followed ICU current logic ,  in ICU code they are returning 0 https://github.com/microsoft/icu/blob/master/icu/icu4c/source/i18n/decimfmt.cpp#L1200.

From the docs `In most locales these are the same, but sometimes they are different. For example, if the primary grouping interval is 3, and the secondary is 2, then this corresponds to the pattern "#,##,##0", and the number 123456789 is formatted as "12,34,56,789".` 
So I guess I can just check if they are the same return only primaryGroupingSize, otherwise both of them. Does it sound good?

Reply:
How does the return of `GetLocaleInfoNative(LocaleGroupingData type)` get used? I guess, what are the expectations for that function? Are `[3, 3]` and `[3]` the same? From the microsoft/icu files, it seems like a format like `#,##0.0#;(#,##0.0#)` expects the `secondaryGroupingSize` to be 0. So should we have `[3, 0]` be a possible return value? Or is `[3, 0]` the same as `[3]`?

Is there an explanation of what we should expect for different formats somewhere?

Reply:
As a validation I use test cases that we have so behaviour will not change for users . When I return `[3, 0]` it breaks some tests like
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoNumberGroupSizes.cs#L14